integration/update_field_test.go

```diff
@@ -283,112 +283,142 @@ func TestUpdateFieldInc(t *testing.T) {
-				filter:   bson.D{{"_id", "int64"}},
+				id:       "int64",
 				update:   bson.D{{"$inc", bson.D{{"v", int32(1)}}}},
 				expected: bson.D{{"_id", "int64"}, {"v", int64(43)}},
 			},
 			"LongIncrement": {
-				filter:   bson.D{{"_id", "int64"}},
+				id:       "int64",
 				update:   bson.D{{"$inc", bson.D{{"v", int64(1)}}}},
 				expected: bson.D{{"_id", "int64"}, {"v", int64(43)}},
 			},
 			"LongNegativeIncrement": {
-				filter:   bson.D{{"_id", "int64"}},
+				id:       "int64",
 				update:   bson.D{{"$inc", bson.D{{"v", int64(-1)}}}},
 				expected: bson.D{{"_id", "int64"}, {"v", int64(41)}},
 			},
 			"LongIncrementDoubleField": {
-				filter:   bson.D{{"_id", "double"}},
+				id:       "double",
 				update:   bson.D{{"$inc", bson.D{{"v", int64(1)}}}},
 				expected: bson.D{{"_id", "double"}, {"v", float64(43.13)}},
 			},
 			"LongIncrementIntField": {
-				filter:   bson.D{{"_id", "int32"}},
+				id:       "int32",
 				update:   bson.D{{"$inc", bson.D{{"v", int64(1)}}}},
 				expected: bson.D{{"_id", "int32"}, {"v", int64(43)}},
 			},
 
 			"FieldNotExist": {
-				filter:   bson.D{{"_id", "int32"}},
+				id:       "int32",
 				update:   bson.D{{"$inc", bson.D{{"foo", int32(1)}}}},
 				expected: bson.D{{"_id", "int32"}, {"v", int32(42)}, {"foo", int32(1)}},
 			},
 			"IncTwoFields": {
-				filter:   bson.D{{"_id", "int32"}},
+				id:       "int32",
 				update:   bson.D{{"$inc", bson.D{{"foo", int32(12)}, {"v", int32(1)}}}},
 				expected: bson.D{{"_id", "int32"}, {"v", int32(43)}, {"foo", int32(12)}},
 			},
+			"DotNotationDocumentFieldExist": {
+				id:       "document-nested",
+				update:   bson.D{{"$inc", bson.D{{"foo.bar.baz", int32(1)}}}},
+				expected: bson.D{{"_id", "document-nested"}, {"foo", bson.D{{"bar", bson.D{{"baz", int32(2)}}}}}},
+			},
+			"DotNotationDocumentFieldNotExist": {
+				id:       "int32",
+				update:   bson.D{{"$inc", bson.D{{"foo.bar.baz", int32(1)}}}},
+				expected: bson.D{{"_id", "int32"}, {"v", int32(42)}, {"foo", bson.D{{"bar", bson.D{{"baz", int32(1)}}}}}},
+			},
+			"DotNotationArrayFieldExist": {
+				id:       "array-nested",
+				update:   bson.D{{"$inc", bson.D{{"foo.bar.0.baz", int32(1)}}}},
+				expected: bson.D{{"_id", "array-nested"}, {"foo", bson.D{{"bar", bson.A{bson.D{{"baz", int32(2)}}}}}}},
+			},
+			"DotNotationArrayFieldNotExist": {
+				id:     "int32",
+				update: bson.D{{"$inc", bson.D{{"foo.bar.0.baz", int32(1)}}}},
+				expected: bson.D{
+					{"_id", "int32"},
+					{"v", int32(42)},
+					{"foo", bson.D{{"bar", bson.D{{"0", bson.D{{"baz", int32(1)}}}}}}},
+				},
+			},
 		} {
 			name, tc := name, tc
 			t.Run(name, func(t *testing.T) {
 				t.Parallel()
 				ctx, collection := setup.Setup(t, shareddata.Scalars, shareddata.Composites)
 
-				_, err := collection.UpdateOne(ctx, tc.filter, tc.update)
+				_, err := collection.InsertMany(ctx, []any{
+					bson.D{{"_id", "document-nested"}, {"foo", bson.D{{"bar", bson.D{{"baz", int32(1)}}}}}},
```

Comment:
Why not add `document-nested` as `document-embedded` into `shareddata.Composites`?

Reply:
I rewrote tests to use composite data.